plugins/nginx-vhosts/internal-functions

```diff
@@ -22,6 +22,13 @@ fn-nginx-vhosts-manage-hsts() {
   local NGINX_HSTS_CONF="$DOKKU_ROOT/$APP/nginx.conf.d/hsts.conf"
   local HSTS_TEMPLATE="$PLUGIN_AVAILABLE_PATH/nginx-vhosts/templates/hsts.conf.sigil"
 
+  # if hsts is disabled globally then only apply it if an app has it set explicitly
+  if [[ "$(fn-plugin-property-get-default "nginx" "--global" "hsts" "true")" = "false" ]] && \
+     [[ "$(fn-plugin-property-get-default "nginx" "$APP" "hsts" "false")" = "false" ]]; then
+    rm -rf "$NGINX_HSTS_CONF"
+    return;
+  fi
+
```

Comment:
This addition should become a new function:

```
fn-nginx-vhosts-hsts-enabled() {
  declare APP="$1"

  if [[ "$(fn-plugin-property-get-default "nginx" "$APP" "hsts" "")" == "true" ]]; then
    echo "true"
    return
  fi

  fn-plugin-property-get-default "nginx" "--global" "hsts" "true"
}
```

And then the `local HSTS` line near the beginning of `fn-nginx-vhosts-manage-hsts` should be:

```
local HSTS="$(fn-nginx-vhosts-hsts-enabled "$APP")"
```

Reply:
this would break bc.. current default for apps that have never set the property is true

the PR was to allow an option to opt out of the global hsts that is applied

moving it to a reusable function is great, but does not address the issue

Reply:
If the property has never been set, the global value is defaulted to true, so there is no change in the internal workings of Dokku. The only bc-break would be in the report output.

Reply:
If a user wants to opt out of the system's default of enabling hsts, they can set the global value to false. The apps won't have an implicit value of `true` set anymore with the diff I am suggesting, so the system will disable hsts for all apps.

If a user _has_ set a value, then that value would use that value.

I've also updated the diff to reflect that since I'm a noob and realized it would only respect the user's app setting if the app setting was true 😅